OneSignalSDK/onesignal/src/main/java/com/onesignal/LocationController.java

```diff
@@ -58,7 +58,18 @@
    private static boolean locationCoarse;
 
    static final Object syncLock = new Object() {};
-   static LocationHandlerThread locationHandlerThread;
+
+   private static LocationHandlerThread locationHandlerThread;
+   static LocationHandlerThread getLocationHandlerThread() {
+      if (locationHandlerThread == null) {
+         synchronized (syncLock) {
+            if (locationHandlerThread == null)
```

Comment:
Actually this is a cost of the double-checks for synchronized initialization to work ok. In Java, there are different ways to make it thread-safe. One is to make the method synchronized but that will slow performance since we only care about the thread-safe on the initialization.  And the second one is to make thread-safe the initialization, but you know that it can happen that maybe one thread released the lock while other was waiting for it, then it can happen that the thread 1 init the locationHandlerThread while the second thread was waiting to init it, so that's why we need the second check. You can search for double check singleton, which is a common pattern. Let me know if it doesn't make sense 😬

Reply:
ah interesting, ya locking can be heavy. Only can be done a lazy loading singleton pattern where value is `null` like we have here.